### PR TITLE
fix: Fixed camera orientation issue in IOS

### PR DESCRIPTION
**Pull request by tomerh2001**

### CODE REVIEW

**Comment by ismaelsousa** [comment]
I implemented something similar in v2 of the vision camera, does this work fine for you? @tomerh2001

**Comment by tomerh2001** [comment]
> I implemented something similar in v2 of the vision camera, does this work fine for you? @tomerh2001 

I checked the V2 branch, the code there still defaults to .up - without using the frame's orientation, what are you talking about?

If you're asking if my changes worked for me, then yeah

**Comment by ismaelsousa** [comment]
I patched locally, I gonna test your changes here :)

File: ios/VisionCameraOcr.swift [hunk 1]
```diff
@@ -105,23 +111,52 @@ public class OCRFrameProcessorPlugin: NSObject, FrameProcessorPluginBase {
...
+        case .right:
+            return .left
+        case .leftMirrored:
+            return .rightMirrored
+        case .rightMirrored:
+            return .leftMirrored
+        default:
+            return orientation
+        }
+    }
```
Comment (ismaelsousa) [inline ios/VisionCameraOcr.swift:136]: why are you inverting here?

Comment (tomerh2001) [reply ios/VisionCameraOcr.swift:136]: In my testing it seemed MLKit uses the opposite meanings of the orientation reported from frame.orientation

If that isn't replicated for anyone else: maybe a better solution would be to introduce an optional orientation parameter to scanOCR

Comment (ismaelsousa) [reply ios/VisionCameraOcr.swift:136]: Hmm, I guess this is the way of the vision camera is creating the preview

> If that isn't replicated for anyone else: maybe a better solution would be to introduce an optional orientation parameter to scanOCR

that's a good solution indeed

Comment (ismaelsousa) [reply ios/VisionCameraOcr.swift:136]: It looks like not all devices have this behavior

**Comment by ismaelsousa** [comment]
on android we have this https://github.com/mrousavy/react-native-vision-camera/pull/2807

**Pull request closed without merging**